src/include/mpir_hwtopo.h

```diff
@@ -57,6 +57,14 @@ bool MPIR_hwtopo_is_initialized(void);
  */
 MPIR_hwtopo_gid_t MPIR_hwtopo_get_leaf(void);
 
+/*
+ * Return the local id for the object with global id: gid. The local
+ * id corresponds to the hwloc's logical index of objects at a certain
+ * level of the topology tree. This means that if there are N objects
+ * at one level, the logical index is an integer in the range [0:N)
+ */
+int MPIR_hwtopo_get_lid(MPIR_hwtopo_gid_t gid);
+
```

Comment:
@hzhou I am not sure you will like this but I wonder if it is ok for now as reimplementing the topotree code to make it hwloc logical index independent is much more work